packages/gatsby/src/schema/node-model.js

```diff
@@ -312,23 +312,15 @@ class LocalNodeModel {
       runQueryActivity.end()
     }
 
-    let trackInlineObjectsActivity
-    if (tracer) {
-      trackInlineObjectsActivity = reporter.phantomActivity(
-        `trackInlineObjects`,
-        {
-          parentSpan: tracer.getParentActivity().span,
-        }
-      )
-      trackInlineObjectsActivity.start()
-    }
-
-    entries.forEach(node => this.trackInlineObjectsInRootNode(node))
-
-    if (trackInlineObjectsActivity) {
-      trackInlineObjectsActivity.end()
+    return {
+      gqlType,
+      entries: entries.map(node => {
+        // With GatsbyIterable it happens lazily as we iterate
+        this.trackInlineObjectsInRootNode(node)
+        return node
+      }),
+      totalCount,
```

Comment:
Behavior change: switching from eager to lazy tracking of inline objects during iteration.

Now `entries` is an instance of `GatsbyIterable`, so we can apply this lazily. This makes a difference for `lmdb` because each iteration is essentially a separate query (reset cursor, re-apply filters, etc, which can be slow).